recipes/libpqxx/all/patches/7.3.1.patch

```diff
@@ -0,0 +1,247 @@
+diff --git a/include/pqxx/binarystring.hxx b/include/pqxx/binarystring.hxx
+index 780cd6ec..433854ca 100644
+--- a/include/pqxx/binarystring.hxx
++++ b/include/pqxx/binarystring.hxx
+@@ -65,7 +65,7 @@ public:
+   using const_iterator = const_pointer;
+   using const_reverse_iterator = std::reverse_iterator<const_iterator>;
+ 
+-  [[deprecated("Use std::byte for binary data.")]] binarystring(
++  PQXX_DEPRECATED("Use std::byte for binary data.") binarystring(
+     binarystring const &) = default;
+ 
+   /// Read and unescape bytea field.
+@@ -73,22 +73,22 @@ public:
+    * isn't.
+    * @param F the field to read; must be a bytea field
+    */
+-  [[deprecated("Use std::byte for binary data.")]] explicit binarystring(
++  PQXX_DEPRECATED("Use std::byte for binary data.") explicit binarystring(
+     field const &);
+ 
+   /// Copy binary data from std::string_view on binary data.
+   /** This is inefficient in that it copies the data to a buffer allocated on
+    * the heap.
+    */
+-  [[deprecated("Use std::byte for binary data.")]] explicit binarystring(
++  PQXX_DEPRECATED("Use std::byte for binary data.") explicit binarystring(
+     std::string_view);
+ 
+   /// Copy binary data of given length straight out of memory.
+-  [[deprecated("Use std::byte for binary data.")]] binarystring(
++  PQXX_DEPRECATED("Use std::byte for binary data.") binarystring(
+     void const *, std::size_t);
+ 
+   /// Efficiently wrap a buffer of binary data in a @c binarystring.
+-  [[deprecated("Use std::byte for binary data.")]] binarystring(
++  PQXX_DEPRECATED("Use std::byte for binary data.") binarystring(
+     std::shared_ptr<value_type> ptr, size_type size) :
+           m_buf{std::move(ptr)}, m_size{size}
+   {}
+diff --git a/include/pqxx/compiler-public.hxx b/include/pqxx/compiler-public.hxx
+index 08d7ac9a..d5d49477 100644
+--- a/include/pqxx/compiler-public.hxx
++++ b/include/pqxx/compiler-public.hxx
+@@ -43,6 +43,11 @@
+ // Workarounds for Microsoft Visual C++
+ #  ifdef _MSC_VER
+ 
++// Workarounds for deprecated attribute syntax error in Visual Studio 2017.
++#    if _MSC_VER < 1920
++#      define PQXX_DEPRECATED(MESSAGE) __declspec(deprecated( #MESSAGE ))
++#    endif
++
```

Comment:
Have you made this patch available to upstream?

Reply:
Yes! It was already merged into master.
https://github.com/jtv/libpqxx/pull/406